subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/suite/filter/FilterFactoryImpl.kt

```diff
@@ -80,26 +85,33 @@ internal class FilterFactoryImpl(
     private fun MutableList<TestsFilter>.addSourcePreviousSignatureFilters() {
         val previousStatuses = filterData.fromRunHistory.previousStatuses
         if (previousStatuses.included.isNotEmpty() || previousStatuses.excluded.isNotEmpty()) {
-            val previousRunTests = factoryLegacy
+
+            factoryLegacy
                 .createReadReport(legacyReportConfig)
                 .getTests()
-                .getOrThrow()
-            if (previousStatuses.included.isNotEmpty()) {
-                add(
-                    IncludeByTestSignaturesFilter(
-                        source = TestsFilter.Signatures.Source.PreviousRun,
-                        signatures = previousRunTests.filterBy(previousStatuses.included)
-                    )
-                )
-            }
-            if (previousStatuses.excluded.isNotEmpty()) {
-                add(
-                    ExcludeByTestSignaturesFilter(
-                        source = TestsFilter.Signatures.Source.PreviousRun,
-                        signatures = previousRunTests.filterBy(previousStatuses.excluded)
-                    )
+                .fold(
+                    onSuccess = { previousRunTests ->
+                        if (previousStatuses.included.isNotEmpty()) {
+                            add(
+                                IncludeByTestSignaturesFilter(
+                                    source = TestsFilter.Signatures.Source.PreviousRun,
+                                    signatures = previousRunTests.filterBy(previousStatuses.included)
+                                )
+                            )
+                        }
+                        if (previousStatuses.excluded.isNotEmpty()) {
+                            add(
+                                ExcludeByTestSignaturesFilter(
+                                    source = TestsFilter.Signatures.Source.PreviousRun,
+                                    signatures = previousRunTests.filterBy(previousStatuses.excluded)
+                                )
+                            )
+                        }
+                    },
+                    onFailure = { throwable ->
+                        logger.info("Can't get tests from previous run: ${throwable.message}")
```

Comment:
warning

Reply:
we have three cases:
1. ok -> apply filter
2. no report found -> info
3. network problem -> warning

i decided to merge 2 and 3 with info level log, because i don't think we want to react on this somehow, and it is for simplicity mainly

warning now for 2+3 i think is not a good alternative

Do you want me to make it 3 cases result?

Reply:
info now create only noise, we could simply do nothing comparing to `log.info`

Reply:
i don't think so;
it will help us to understand what's going on in build by looking for this message